be/src/fs/fs_s3.cpp

```diff
@@ -307,19 +340,58 @@ static std::shared_ptr<Aws::S3::S3Client> new_s3client(const S3URI& uri, const F
             config.maxConnections = config::object_storage_max_connection;
         }
     } else {
-        if (!uri.endpoint().empty()) {
-            config.endpointOverride = uri.endpoint();
+        // resolve endpoint
+        auto it = opts._fs_options.find("fs.s3a.endpoint");
+        if (it != opts._fs_options.end() && !it->second.empty()) {
+            config.endpointOverride = it->second;
         } else if (!config::object_storage_endpoint.empty()) {
             config.endpointOverride = config::object_storage_endpoint;
+        }
+        // resolve region
+        it = opts._fs_options.find("fs.s3a.endpoint.region");
+        if (it != opts._fs_options.end() && !it->second.empty()) {
+            config.region = it->second;
+        } else if (!config::object_storage_region.empty()) {
+            config.region = config::object_storage_region;
+        }
+        // resolve ssl
+        it = opts._fs_options.find("fs.s3a.connection.ssl.enabled");
+        if (it != opts._fs_options.end()) {
+            if (it->second.compare("true") == 0) {
+                config.scheme = Aws::Http::Scheme::HTTPS;
+            } else {
+                config.scheme = Aws::Http::Scheme::HTTP;
+            }
         } else if (config::object_storage_endpoint_use_https) {
             config.scheme = Aws::Http::Scheme::HTTPS;
         } else {
             config.scheme = Aws::Http::Scheme::HTTP;
         }
-        if (!config::object_storage_region.empty()) {
-            config.region = config::object_storage_region;
-        }
         config.maxConnections = config::object_storage_max_connection;
+        // resolve retry
+        int64_t s3client_max_retries = 5;
+        it = opts._fs_options.find("fs.s3a.retry.limit");
+        if (it != opts._fs_options.end() && !it->second.empty()) {
+            s3client_max_retries = stoi(it->second);
+        }
+        int64_t s3client_retry_scale_factor = 200;
+        it = opts._fs_options.find("fs.s3a.retry.interval");
+        if (it != opts._fs_options.end() && !it->second.empty()) {
+            s3client_retry_scale_factor = stoi(it->second);
```

Comment:
need to hand `std::stoi` exception when provided interval is an invalid integer or negative integer.

Reply:
OK